### PR TITLE
Init lockdown api v2

### PR DESCRIPTION
**Pull request by maingockien01**
Issue#376 Start API v2
This pull request:
- removed the old api v1 
- replaced with new framework NestJS, and Typescript language
- added api endpoint for datasetentries (the return data has not met the requirement). 
- had a rough README document.

### CODE REVIEW

**Comment by jeff-knurek** [comment]
:+1: thanks for getting a quick start on this.

Some quick thoughts before i look too deep into this.

* I'd like to not have a structure that is `API/src/src` and instead just the one level to source code
* that then means the current API/README would be updated directly 
* the `gitignore` contains some things (like IDE settings) that should go to the top project level `gitignore` file 
* some of the files removed still have potential value, _(for example: `API/src/data/population.json` as well as the config.js)_ I don't know how much value though so maybe they can be removed, i guess I just want to raise awareness

**Comment by maingockien01** [comment]
@jeff-knurek 
As we discussed last night, I made the PR for now.
The api_v1 source code is kept in folder Reference now.  
We can move docs later when you figured out where you want to put them. 
Unit test files will be in the same folder as source files. I updated the command for unit test in package.json.

File: API/.gitignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+lerna-debug.log*
+
+# OS
+.DS_Store
```
Comment (jeff-knurek) [inline API/.gitignore:14]: can you move these OS & IDE specific things to the top level `.gitignore` file

**Review by jeff-knurek — COMMENTED** [review]
sorry for the delay @maingockien01 - one small change to the gitignore please, and another request

can you add a file called `sonar-project.properties` at the top level, with the following contents. 

```
# Define the root directory(s) for main sources and tests
sonar.sources=MAP,API
# Exclude files in a test directory from the main source files
sonar.exclusions=API/Reference/api_v1/**/*
```

**Comment by maingockien01** [comment]
@jeff-knurek let me know if it is able to merge now. Then, I could make one more small commit to clean files.

File: .sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Add data about the pull request
+sonar.pullrequest.key=379
+sonar.pullrequest.branch=api_v2
+sonar.pullrequest.base=master
```
Comment (jeff-knurek) [inline .sonar-project.properties:4]: sonarcloud already know about Github info, so it's not needed to be saved 

in fact, using a `.sonar-project.properties` is not advised for SonarCloud while we using the Automatic Analysis https://sonarcloud.io/documentation/analysis/automatic-analysis/#presence-of-a-properties-file

File: .sonarcloud.properties [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Define the root directory(s) for main sources and tests
+sonar.sources=MAP,API/src, API/src
```
Comment (jeff-knurek) [inline .sonarcloud.properties:2]: `API/src` is duplicated, should only have it once

File: .sonarcloud.properties [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+# Exclude files in a test directory from the main source files
+sonar.exclusions=**/Reference/api_v1/**/*, API/Reference/api_v1/src/**/*
```
Comment (jeff-knurek) [inline .sonarcloud.properties:4]: these paths also overlap and might be causing the error, try taking one of them out

File: API/.gitignore (hunk 1 above)
Comment (jeff-knurek) [reply API/.gitignore:14]: looks like these configs are in the top level now, and should be removed from this file :pray:

File: API/sonar-project.properties [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Define the root directory(s) for main sources and tests
+sonar.sources=src
```
Comment (jeff-knurek) [inline API/sonar-project.properties:2]: I don't think this file provides any value :thinking:

File: sonar-project.properties [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Add data about the pull request
+sonar.pullrequest.key=379
+sonar.pullrequest.branch=api_v2
+sonar.pullrequest.base=master
```
Comment (jeff-knurek) [inline sonar-project.properties:4]: also, this file can be removed

File: .sonar-project.properties (hunk 2 above)
Comment (maingockien01) [reply .sonar-project.properties:4]: I'll remove .sonar-project.properties and only keep .sonarcloud.properties

File: .sonarcloud.properties (hunk 3 above)
Comment (maingockien01) [reply .sonarcloud.properties:2]: done!

File: .sonarcloud.properties (hunk 4 above)
Comment (maingockien01) [reply .sonarcloud.properties:4]: done!

File: .sonar-project.properties (hunk 2 above)
Comment (jeff-knurek) [reply .sonar-project.properties]: success!! 
now there's only 1 issue being reported from the new work you've done: https://sonarcloud.io/project/issues?id=TheIOFoundation_ProjectLockdown&open=AXmAEfllz5W-InxO1660&pullRequest=379&resolved=false&types=CODE_SMELL
you can click `Why is this an issue?` to understand a bit more of, well, why ;)

**Review by jeff-knurek — APPROVED** [review]

**Pull request merged**